### ISSUE TITLE
OverloadMethodDeclarationOrder check ignores anything besides methods

### ISSUE THREAD

**Issue by Zopsss** [checkstyle/checkstyle#14765]
I have read check documentation: https://checkstyle.org/checks/coding/overloadmethodsdeclarationorder.html
I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

The [OverloadMethodDeclarationOrderCheck](https://checkstyle.org/checks/coding/overloadmethodsdeclarationorder.html) gives us error if there is another method between the overloaded methods. But it does not gives error when there is anything else besides method between overloaded methods like a variable, array, inner interface, inner class, etc.

This is how the check works currently:

#### Test.java
```java
public class Test {
    void test() {}

    void test(int x) {}

    private interface Testing {}

    String str;

    void test(String str) {}
}
```

#### config.xml
```xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
        "https://checkstyle.org/dtds/configuration_1_3.dtd">

<module name="Checker">
    <module name="TreeWalker">
        <module name="OverloadMethodsDeclarationOrder"/>
    </module>
</module>
```

#### Output
```
➜ java -jar ../checkstyle-10.15.0-all.jar -c config.xml Test.java
Starting audit...
Audit done.
```

The check should be giving error for the last method
```java
void test(String str) {}
```

---

**Describe what you expect in detail.**

The check should be give an error something like this for the above example:

```
➜ java -jar ../checkstyle-10.15.0-all.jar -c config.xml Test.java
Starting audit...
[ERROR] C:\checkstyle testing\OverloadMethods\Test.java:10:5: 
All overloaded methods should be placed next to each other. Placing non-overloaded methods
 in between overloaded methods with the same type is a violation. 
Previous overloaded method located at line '4'. [OverloadMethodsDeclarationOrder]
Audit done.
Checkstyle ends with 1 errors.
```

---

**Comment by Zopsss** [comment]
I'm working on it!

**Comment by nrmancuso** [comment]
Closed via #14766
